packages/compiler/core/parser.ts

```diff
@@ -964,6 +979,17 @@ export function parse(code: string | SourceFile, options: ParseOptions = {}): Ca
     return expr;
   }
 
+  function parseStandaloneReferenceExpression() {
+    const expr = parseReferenceExpression();
+    if (
+      expr.kind === SyntaxKind.TypeReference &&
+      parseDiagnostics.length === 0 &&
```

Comment:
Why do we not report this with other parse diagnostics?

Reply:
So if there is already a parse error while trying to parse a type reference, I don't want to emit another error because we are not at the end of the "file". THis would create a double error everytime you have an invalid type reference

Reply:
Oh I see. We might not have reached the end of the file because we didn't consume all the tokens when emitting the error.

Reply:
exactly